src/odemis/driver/technolution.py

```diff
@@ -1015,6 +991,11 @@ def __init__(self, name, role, parent, **kwargs):
 
         # The minimum of the cell resolution cannot be lower than the minimum effective cell size.
         self.cellCompleteResolution = model.ResolutionVA((900, 900), ((12, 12), (1000, 1000)))
+        self.cellCompleteResolution.subscribe(self._updateFrameDuration)
+        self.parent._ebeam_scanner.dwellTime.subscribe(self._updateFrameDuration)
+        # acquisition time for a single field image including overscanned pixels and flyback time
+        acq_time = self.getTotalFieldScanTime()
```

Comment:
Move this *after* creating `.frameDuration`, as  `_updateFrameDuration()` needs this attribute (so it's a little clearer). Also, you can avoid computing `acq_time` explicitly: initialize at 0, and in add an `init=True` in one of the subscribers.